corehq/apps/userreports/models.py

```diff
@@ -822,6 +824,16 @@ def _get_datasource_config(cls, static_config, domain):
         doc['_id'] = cls.get_doc_id(domain, doc['table_id'])
         return DataSourceConfiguration.wrap(doc)
 
+    def validate_db_config(self):
```

Comment:
This should be in `DataSourceConfiguration`

Reply:
See https://github.com/dimagi/commcare-hq/pull/24046#discussion_r276514388